packages/datadog_grpc_interceptor/test/datadog_grpc_interceptor_test.dart

```diff
@@ -130,87 +249,80 @@ void main() {
       expect(attributes['_dd.span_id'], isNull);
     });
 
-    test('Interceptor passes on proper metadata', () async {
+    test('extracts b3m headers and sets attributes', () async {
       when(() => mockDatadog.isFirstPartyHost(any())).thenReturn(true);
 
       final interceptor = DatadogGrpcInterceptor(mockDatadog, channel);
 
       final stub = GreeterClient(channel, interceptors: [interceptor]);
+      // Randomly generated
+      //  - 61ffb765f4f77e3b == 7061564389269667387
+      //  - 3eb5c1bcb46ab916 == 4518730817361066262
+      final options = CallOptions(metadata: {
+        'x-b3-traceid': '000000000000000061ffb765f4f77e3b',
```

Comment:
I remember seeing somewhere that some implementations might be case sensitive. So it should be both `X-B3-TradeId` and value should use capital letters `000000000000000061FFB765F4F77E3B` and so on.

Reply:
For grpc specifically, the headers need to be lowercase (I think because all gRPC metadata fields are lowercased)
https://github.com/openzipkin/b3-propagation#grpc-encoding

I'll fix the hex IDs to generate capital letters.

Reply:
Yes you're - sorry for not double checking on my side.